### DISCUSSION TITLE
Passing multiple Parameters to `mutationFn`

### DISCUSSION THREAD

**Discussion by premdasvm** [TanStack/query#6101] [Q&A] [answered]
Hi,

Is it possible to pass multiple parameters to the `mutationFn` ?

something like

```ts
const { isLoading: isEditLoading, mutateAsync: customerUpdate } = useMutation({
    mutationFn: (id, data) => updateCustomer(id, data),
    onSuccess: () => {
        customerDialog.value = false;
        toast.add({ severity: 'success', summary: 'Customer updated', detail: 'Customer update successful' })
        refetch()
    },
    onError: () => {
    },
}

await customerUpdate(1, { name: 'Test'} ) ;

```
i ended up passing the `id` as part of `data` it self, which works. i just wanted to make sure that it was the right way.
i'm using `vue-query` by the way.

**Comment by TkDodo** [comment, marked as answer]
> Is it possible to pass multiple parameters to the mutationFn ?

no. see: https://tkdodo.eu/blog/mastering-mutations-in-react-query#mutations-only-take-one-argument-for-variables

**Reply by premdasvm** [reply]
@TkDodo Thanks for the response. That was really helpful.

i've updated my `useMutation` to

```ts
const { mutateAsync: customerUpdate } = useMutation({
    mutationFn: ({ id, variables }: { id: number, variables: UpdateCustomerType }) => updateCustomer(id, variables),
    onSuccess: () => {},
    onError: () => {},
})

await editUserMethod({ id: 1, variables: { name: 'test' } });

```
and now it's working as i expected.
